third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -1313,10 +1302,26 @@ extern "C" void V8RecordReplayRegisterBrowserEventCallback(
   void (*callback)(const char* name, const char* payload)
 );
 
+static bool TestEnv(const char* env) {
+  const char* v = getenv(env);
+  return v && v[0] && v[0] != '0';
+}
+
 void SetupRecordReplayCommands(v8::Isolate* isolate) {
   V8RecordReplaySetAPIObjectIdCallback(GetAPIObjectIdCallback);
   V8RecordReplayRegisterBrowserEventCallback(HandleBrowserEvent);
 
+  gCollectSourceMaps = !TestEnv("RECORD_REPLAY_DISABLE_SOURCEMAP_COLLECTION");
+
+  // When we aren't collecting source maps and are recording we don't need the
+  // record/replay script for anything, so we don't compile and run it. The script
+  // adds significant overhead, mainly due to enabling V8 inspector features.
```

Comment:
It's good I read this patch.  The network monitor as currently drafted uses this javascript code as a bridging mechanism between the events coming in from chrome and being sent out to the linker via `OnNetworkRequest(Event)?`.

If we're avoiding JS execution that will need to be reworked a bit.